### PR TITLE
Add Definition/Doc base rendering

### PR DESCRIPTION
**Pull request by hojberg**
## Overview
Add Definition/Doc with a bare bones (un-styled) rendering of Unison
Docs.

part of https://github.com/unisonweb/codebase-ui/issues/77

## Implementation notes
This mostly follows the Haskell type from the backend with a few minor
changes to make it fit within the frontend (like Reference, Source, and
Syntax).

Unrelated, this also deletes `src/Source.elm` that was completely unused (had been replaced by `src/Definition/Source.elm`).

## Interesting/controversial decisions
I decided to use extensible records to avoid a circular dependency with regards to `TermDetail` and `TypeDetail`. An alternative could be to move those definitions into `WorkspaceItem` where they are used and combined with `Doc`.

## Loose ends
* This PR just includes the structure of the html and the rendering remains un-styled.
* Changing fold state is not yet implemented.

### CODE REVIEW

**Comment by hojberg** [comment]
Example (GitHub wouldn't let me add this to the description of the PR after the fact)

Screen Shot 2021-06-17 at 10 23 25

File: src/Definition/Doc.elm [hunk 1]
```diff
@@ -0,0 +1,395 @@
...
+                UntitledSection content ->
+                    section [] (List.map viewAtCurrentSectionLevel content)
+
+                Column content ->
+                    div [ class "column" ] (List.map viewAtCurrentSectionLevel content)
+
+                Group content ->
+                    span [ class "group" ] [ viewAtCurrentSectionLevel content ]
+    in
+    div [ class "doc" ] [ view_ 0 document ]
```
Comment (pchiusano) [inline src/Definition/Doc.elm:386]: Should this be `view_ 1 document`? Since when you encounter a section you emit `hN` where `N` is the section level.

**Review by pchiusano — APPROVED** [review]
Nice! Left one question for you. 

It's possible there'll be some tweaks to this if there's backend changes to the `Doc` type being produced, but that should be easy to work out.

File: src/Definition/Doc.elm (hunk 1 above)
Comment (hojberg) [reply src/Definition/Doc.elm:386]: yeah you're right—good catch!

**Pull request merged**